core/state/database.go

```diff
@@ -124,6 +124,10 @@ type Trie interface {
 	// The returned map could be nil if the witness is empty.
 	Witness() map[string]struct{}
 
+	// WitnessPaths returns a set of paths for all trie nodes. For future reference,
+	// witness can be deprecated and used as a replacement to witness.
```

Comment:
```suggestion
	// witness paths can be deprecated and used as a replacement to witness.
```

Reply:
I disagree with this comment, because the witness will be spec'd by something exterior and this external consumer will not be interested in the paths because of size reasons. But we shall see.